### PR TITLE
fixup! Wait for presto user after rpm installation

### PR DESCRIPTION
**Pull request by cawallin**
Fix the unit test so that it doesn't hang forever in checking for the Presto user.

### CODE REVIEW

**Comment by cawallin** [comment]
FYI @kokosing

**Comment by rschlussel** [comment]
lgtm

File: tests/unit/test_server.py [hunk 1]
```diff
@@ -53,23 +53,23 @@ def test_install_server(self, mock_execute, mock_check_rpm):
...
 
+    @patch('prestoadmin.server.sudo')
     @patch('prestoadmin.server.package.deploy_install')
     @patch('prestoadmin.server.update_configs')
-    def test_deploy_install_configure(self, mock_update, mock_install):
+    def test_deploy_install_configure(self, mock_update, mock_install,
+                                      mock_sudo):
         local_path = "/any/path/rpm"
+        mock_sudo.side_effect = self.mock_failed_then_succeeded()
         server.deploy_install_configure(local_path)
```
Comment (petroav) [inline tests/unit/test_server.py]: Add a newline after this to separate the test from the assertion statements.

File: tests/unit/test_server.py [hunk 2]
```diff
@@ -53,23 +53,23 @@ def test_install_server(self, mock_execute, mock_check_rpm):
...
         mock_install.assert_called_with(local_path)
         mock_update.assert_called_with()
```
Comment (petroav) [inline tests/unit/test_server.py]: If we're not checking the arguments used to call the mock maybe we can substitute this call with something simpler?

File: tests/unit/test_server.py [hunk 3]
```diff
@@ -454,3 +454,10 @@ def test_multiple_version_rpms(self, mock_run):
             call('rpm -q --qf \"%{VERSION}\\n\" presto-server-rpm'),
         ])
         self.assertEqual(expected, '')
+
+    def mock_failed_then_succeeded(self):
```
Comment (petroav) [inline tests/unit/test_server.py]: I prefer `mock_fail_then_succeed`. Your call.

**Comment by petroav** [comment]
looks good

File: tests/unit/test_server.py (hunk 1 above)
Comment (cawallin) [inline tests/unit/test_server.py]: Done

File: tests/unit/test_server.py (hunk 3 above)
Comment (cawallin) [inline tests/unit/test_server.py]: Sure, changed it to "succeeded" to match the attribute, but that's fine.

**Pull request merged**